libraries/Microsoft.Bot.Builder.Core.Extensions/ActivityFilterMiddleware.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Core.Extensions
+{
+    public class ActivityFilterMiddleware : IMiddleware
+    {
+        /// <summary>
+        /// The type of Activity that the middleware should check for. e.g. ConversationUpdated or Message
+        /// </summary>
+        private readonly string _activityType;
+
+        /// <summary>
+        /// Handler to call when a matching Activity type is received
+        /// </summary>
+        private readonly ActivityFilterHandler _activityFilterHandler;
+
+        /// <summary>
+        /// Middleware to call when a matching Activity type is received
+        /// </summary>
+        private readonly IMiddleware _nextMiddleware;
+
+        public ActivityFilterMiddleware(string activityType, ActivityFilterHandler handler)
```

Comment:
>ActivityFilterHandler [](start = 61, length = 21)

I don't see the need for this.  Isn't this just the middleware pipeline?

Reply:
@tomlm Maybe the naming is off here.  The point of this piece of middleware is to allow the developer to take action depending on the incoming activity type. e.g. If there is a conversation update then they might want to send a message back to the user (e.g. a welcome message), similar to the method in the v3 controllers which handled various different activity types. @cleemullins

Reply:
@tomlm for additional context, this is a C# version of the middleware @Stevenic